holoviews/core/spaces.py

```diff
@@ -667,9 +680,19 @@ def clone(self, data=None, shared_data=True, new_type=None, *args, **overrides):
         """
         if data is None and shared_data:
             data = self.data
-        return super(UniformNdMapping, self).clone(overrides.pop('callback', self.callback),
-                                                   shared_data, new_type,
-                                                   *(data,) + args, **overrides)
+        clone = super(UniformNdMapping, self).clone(overrides.pop('callback', self.callback),
+                                                    shared_data, new_type,
+                                                    *(data,) + args, **overrides)
+
+        # Ensure the clone references this object to ensure
+        # stream sources are inherited
+        if clone.callback is self.callback:
+            clone.callback = self.callback.clone()
```

Comment:
I don't quite get this line. Why would ``self.callback`` have a clone method? Isn't that a ``Callable`` around the callback?